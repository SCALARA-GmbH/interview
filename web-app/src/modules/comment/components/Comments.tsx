import React, { useEffect, useState } from 'react';
import axios from 'axios';
import { useForm } from 'react-hook-form';
import CustomInput from '../../../shared/components/Input';
import Box from '@mui/material/Box';
import Grid from '@mui/material/Grid';
import CustomButton from '../../../shared/components/Button';
import CustomGridItem from '../../../shared/components/GridItem';
import { useParams } from 'react-router-dom';
export interface Comment {
  date: Date;
  content: string;
  postId: number;
}

export interface CommentsDto {
  date?: Date;
  content: string;
  postId: number;
}

const Comments: React.FunctionComponent = () => {
  const [comments, setComments] = useState<Comment[]>([]);
  const { topicId }: any = useParams();
  const { register, reset, handleSubmit: handleFormSubmit, formState: { errors } } = useForm<CommentsDto>();
  const createPost = async (data: CommentsDto) => {
    await axios.post<CommentsDto>('http://localhost:5000/comments', data);
  };
  const handleSubmit = async (data: CommentsDto) => {


    data.postId = 1;
    await createPost(data);
    reset({ content: '' });
    await fetchComments(1);
  };

  const fetchComments = async (id: number) => {
    const result = await axios.get<Comment[]>('http://localhost:5000/comments?post-id=' + id);
    setComments(result?.data ? result?.data : []);
  };

  useEffect(() => {
    console.log(topicId)
    fetchComments(topicId);
  }, []);
  return (
    <>

      <Box sx={{ flexGrow: 1 }}>
        <Grid container columns={12}>
          <Grid item xs={2}>

          </Grid>
          <Grid item xs={7}>
            <CustomGridItem>
              <div>
                <h1>Comments</h1>


                <Box
                  component="form"
                  noValidate
                  sx={{
                    display: 'grid',
                    gridTemplateColumns: { sm: '1fr 1fr' },
                    gap: 2,
                  }}
                >
                  {/* <input placeholder={'Post a comment'} {...register('content', { required: true })} /> */}
                  <CustomInput defaultValue="Post a comment..." id="bootstrap-input" {...register('content', { required: true })} />
                  {errors.content && <div className="err"><span className="red-text">*</span>This field is required</div>}

                  <CustomButton variant="contained" onClick={handleFormSubmit(handleSubmit)}>Post a Comment</CustomButton>
                </Box>


                {comments.map((comment, index) =>
                  <div key={index}>
                    <h2>{comment.content}</h2>
                    {comment.date}
                  </div>
                )}
              </div>
            </CustomGridItem>
          </Grid>
          <Grid item xs={3}>

          </Grid>
        </Grid>
      </Box>




    </>
  );
};

export default Comments;