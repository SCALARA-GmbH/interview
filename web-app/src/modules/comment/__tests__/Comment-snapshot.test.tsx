import React from 'react';
import renderer from 'react-test-renderer';

import Comments from '../components/Comments';

it('renders correctly ', () => {
  const tree = renderer.create(<Comments />).toJSON();
  expect(tree).toMatchSnapshot();
});