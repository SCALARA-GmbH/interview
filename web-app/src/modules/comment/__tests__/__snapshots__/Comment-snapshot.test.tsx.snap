// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are no items 1`] = `
<div
  className="MuiBox-root css-i9gxme"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-undefined css-11lq3yg-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
    />
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-7 css-klo4dh-MuiGrid-root"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-16d00og-MuiPaper-root"
      >
        <div>
          <h1>
            Comments
          </h1>
          <form
            className="MuiBox-root css-z41f3n"
            noValidate={true}
          >
            <div
              className="MuiInputBase-root MuiInputBase-colorPrimary css-1fq6r3k-MuiInputBase-root"
              onClick={[Function]}
            >
              <input
                className="MuiInputBase-input css-yz9k0d-MuiInputBase-input"
                defaultValue="Post a comment..."
                id="bootstrap-input"
                name="content"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                type="text"
              />
            </div>
            <button
              className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-3qggjx-MuiButtonBase-root-MuiButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              Post a Comment
            </button>
          </form>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-j23p4t-MuiGrid-root"
    />
  </div>
</div>
`;
